src/ContainerDetails.jsx

```diff
@@ -12,6 +12,14 @@ const render_container_state = (container) => {
     return cockpit.format(_("Exited"));
 };
 
+const render_container_published_ports = ({ ports }) => {
+    if (ports)
+        return ports
+                .map(port => cockpit.format(_("$0:$1->$2/$3"), (port.host_ip || '0.0.0.0'), port.host_port, port.container_port, port.protocol))
+                .join(', ');
+    return '';
```

Comment:
`''` is okay, but for the learning experience: The standard way to say "don't render anything" is `null`, not an empty string.

Can this please use a more tasteful arrow and some spaces? Also, I don't think this really needs to be i18n'ed. Thus

    port => cockpit.format("$0:$1 → $2/$3", ...)